perceval/backends/gerrit.py

```diff
@@ -0,0 +1,295 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2015 Bitergia
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+#
+# Authors:
+#   Alvaro del Castillo San Felix <acs@bitergia.com>
+#
+
+'''Gerrit backend for Perseval'''
+
+
+from datetime import datetime
+import json
+import logging
+import os.path
+import re
+import subprocess
+from time import time
+
+from ..backend import Backend, BackendCommand
+from ..cache import Cache
+from ..errors import BackendError, CacheError
+from ..utils import DEFAULT_DATETIME, str_to_datetime
+
+
+class Gerrit(Backend):
+
+    name = "gerrit"
+
+    def __init__(self, user=None, url=None, nreviews=None,
+                 cache=None, **nouse):
+        super().__init__(cache=cache)
+        self.repository = url
+        self.nreviews = nreviews
+        self.number_results = None  # last number of results
+        self.client = GerritClient(self.repository, user, nreviews)
+
+    def fetch_from_cache(self):
+        """Fetch the bugs from the cache.
+
+        It returns the bugs stored in the cache object provided during
+        the initialization of the object. If this method is called but
+        no cache object was provided, the method will raise a `CacheError`
+        exception.
+
+        :returns: a generator of items
+
+        :raises CacheError: raised when an error occurs accesing the
+            cache
+        """
+        if not self.cache:
+            raise CacheError(cause="cache instance was not provided")
+
+        cache_items = self.cache.retrieve()
+
+        for raw_items in cache_items:
+            reviews = Gerrit.parse_reviews(raw_items)
+            for review in reviews:
+                yield review
+
+
+    def fetch(self, from_date=DEFAULT_DATETIME):
+        self._purge_cache_queue()
+
+        last_item = self.client.next_retrieve_group_item()
+        reviews = self._get_reviews(last_item)
+        last_nreviews = len(reviews)
+
+        while reviews:
+            review = reviews.pop(0)
+            last_item += 1
+            updated = datetime.fromtimestamp(review['lastUpdated'])
+            if updated <= from_date:
+                logging.debug("No more updates for %s" % (self.repository))
```

Comment:
As I read on many sites, it is recommended to declare a logger at the top of the module and then, use it for logging instead of using the module because this way it writes on the root logger.
You have to create a logger using `logger = logging.getLogger(__name__)' and then write on it using `logger.debug()`, for instance.